edx-platform/nau-basic/lms/templates/footer.html

```diff
@@ -105,3 +113,81 @@
     <link rel="stylesheet" type="text/css" href="${url}"></link>
   % endfor
 % endif
+
+%if modal_force_display:
+## Pop-up tos
+<div class="modal-container">
+  <div class="modal-content">
+    % if modal_tos_text:
+    ${modal_tos_text | n}
+    <script type="text/javascript">
+    
+    </script>
+
+    % endif
+    <button class="button-modal btn-primary">Accept</button>
+  </div>
+</div>
+
+  %if user.is_authenticated:
+  <% value_preference = get_user_preference(user, name_preference_tos) %>
+
+  <script type="text/javascript">
+
+    const newTosModal = (function () {
+      const valuePreference = '${value_preference}';
+      const urlPreferenceTos = '/api/user/v1/preferences/${user}/${name_preference_tos}';
+      const namePreferenceTos ='${name_preference_tos}';
+      const dateJoined ='${user.date_joined}';
+      const removeWord = namePreferenceTos.replace('update_tos_', '').replaceAll('_', '-') 
```

Comment:
How do you plan to handle the case when `${name_preference_tos}` is empty or does not match what you expect?

Reply:
I changed it, so that it only takes the date that should go at the end in case the value does not exist or does not contain the date at the end it will be false